packages/storage_client/lib/src/storage_file_api.dart

```diff
@@ -91,6 +97,106 @@ class StorageFileApi {
     return (response as Map)['Key'] as String;
   }
 
+  /// Upload a file with a token generated from `createUploadSignedUrl`.
+  ///
+  /// [path] The file path, including the file name. Should be of the format `folder/subfolder/filename.png`. The bucket must already exist before attempting to upload.
+  ///
+  /// [token] The token generated from `createUploadSignedUrl`
+  ///
+  /// [file] The body of the file to be stored in the bucket.
+  Future<String> uploadToSignedUrl(
+    String path,
+    String token,
+    File file, [
+    FileOptions fileOptions = const FileOptions(),
+    int? retryAttempts,
+    StorageRetryController? retryController,
+  ]) async {
+    assert(retryAttempts == null || retryAttempts >= 0,
+        'retryAttempts has to be greater or equal to 0');
```

Comment:
Oh yeah, giving it a default value would be so much better!

Reply:
Actually, I just remembered. There is a [global settings for `retryAttempt`](https://github.com/supabase/supabase-flutter/blob/main/packages/storage_client/lib/src/storage_client.dart#L35), and if the `retryAttempt` value passed to the individual methods are `null`, [it will fall back](https://github.com/supabase/supabase-flutter/blob/main/packages/storage_client/lib/src/storage_file_api.dart#L157) to the global default, so we might need to keep it nullable here.